onnxruntime/python/tools/transformers/models/llama/convert_to_onnx.py

```diff
@@ -0,0 +1,474 @@
+import argparse
+import logging
+import os
+import sys
+import tempfile
+from itertools import chain
+from typing import List
+
+import onnx
+import torch
+from transformers import LlamaConfig, LlamaForCausalLM
+
+sys.path.append(os.path.join(os.path.dirname(__file__), "..", ".."))
+from benchmark_helper import Precision, prepare_environment, setup_logger  # noqa: E402
+from llama_parity import main as parity_check  # noqa: E402
+from onnx_model import OnnxModel  # noqa: E402
+
+from onnxruntime import quantization  # noqa: E402
+
+logger = logging.getLogger("")
+
+
+def get_model_inputs(config: LlamaConfig):
+    batch_size, seq_len = 2, 8
+    input_ids = torch.randint(low=0, high=config.vocab_size, size=(batch_size, seq_len), dtype=torch.int64)
+    attn_mask = torch.randint(low=0, high=2, size=(batch_size, seq_len), dtype=torch.int64)
+    # pos_ids is of shape (batch_size, seq_len)
+    pos_ids = attn_mask.long().cumsum(-1) - 1
+    pos_ids.masked_fill_(attn_mask == 0, 1)
```

Comment:
pos_ids.masked_fill_(pos_ids< 0, 0)

Reply:
Those two lines are from Hugging Face's implementation:
https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py#L863-L868

Reply:
Those two lines are from Hugging Face's implementation:

https://github.com/huggingface/transformers/blob/7dcd8703ef904adc3ac19b47f769879221c33849/src/transformers/models/llama/modeling_llama.py#L752-L757

Reply:
Those two lines are from [Hugging Face's implementation](https://github.com/huggingface/transformers/blob/7dcd8703ef904adc3ac19b47f769879221c33849/src/transformers/models/llama/modeling_llama.py#L752-L757).

Reply:
I see. I think you can remove the line since attention_mask is ones, so the line of masked_fill_ has no effect.

Reply:
Removed `masked_fill_`